profiles/common/modules/features/nexteuropa_newsroom/classes/NewsroomHelper.inc

```diff
@@ -0,0 +1,633 @@
+          }
+        }
+      }
+
+      if (count($menu_items) > 0) {
+        // Favor terms over nodes and lower entity ID in case we have multiple.
+        usort($menu_items, function ($a, $b) {
+          if ($a['router_path'] == 'taxonomy/term/%' && $b['router_path'] == 'node/%') {
+            return -1;
+          }
+          elseif ($a['router_path'] == 'node/%' && $b['router_path'] == 'taxonomy/term/%') {
+            return 1;
+          }
+          else {
+            if ($a['plid'] == $b['plid']) {
+              if ($a['weight'] == $b['weight']) {
+                if ($a['mlid'] == $b['mlid']) {
+                  return 0;
+                }
+                else {
+                  return $a['mlid'] < $b['mlid'] ? -1 : 1;
+                }
+              }
+              else {
+                return $a['weight'] < $b['weight'] ? -1 : 1;
+              }
+            }
+            else {
+              return $a['plid'] < $b['plid'] ? -1 : 1;
+            }
+          }
+        });
+
+        $menu_item_link = array_shift($menu_items);
+        $path = $menu_item_link['link_path'];
+      }
+
+      cache_set($cache_key, $path, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    }
+
+    return $path;
+  }
+
+  /**
+   * Try to resolve each taxonomy term with a corresponding ID to map.
+   *
+   * @param string $field_name
+   *   The mapping field.
+   * @param string $field_value
+   *   The mapping field value.
+   *
+   * @return array
+   *   Taxonomy ids.
+   */
+  public static function getTaxonomyTermFromField($field_name, $field_value, $bundle = NULL) {
+    $field_value = is_array($field_value) ? $field_value : array($field_value);
+    $cache_key = 'tech_field_tid:' . md5($field_name . implode('', $field_value) . $bundle);
+    $cache = cache_get($cache_key, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    if ($cache) {
+      $term_ids = $cache->data;
+    }
+    else {
+      $term_ids = array();
+      $query = new EntityFieldQuery();
+      $query->entityCondition('entity_type', 'taxonomy_term')
+        ->fieldCondition($field_name, 'value', $field_value);
+      if ($bundle) {
+        $query->entityCondition('bundle', $bundle);
+      }
+      $result = $query->execute();
+      if (isset($result['taxonomy_term'])) {
+        $term_ids = array_keys($result['taxonomy_term']);
+      }
+
+      cache_set($cache_key, $term_ids, NEXTEUROPA_NEWSROOM_CACHE_TABLE);
+    }
+    return $term_ids;
+  }
+
+  /**
+   * Get events Terms ids.
+   */
+  public static function getEventsTermsIds() {
+    return NewsroomHelper::getTaxonomyTermFromField('field_newsroom_type',
+      array(
+        'consultation_funding',
+        'events',
+        'agenda',
+      ), NEXTEUROPA_NEWSROOM_TYPE_VOCABULARY);
+  }
+
+  /**
+   * Item types or Collection to group by.
+   *
+   * @return array
+   *   Terms
+   */
+  public static function getGroupTypeTerms($selected = array()) {
+    $cache_key = 'newsroom:group_types:' . implode('-', $selected);
```

Comment:
I have rewritten the code and now we use the common way to retrieve newsroom types, we don't need cache here anymore, because it doesn't give performance improvements.